KyanBar/models/NowPlayingModel.swift

```diff
@@ -7,42 +7,22 @@
 
 import Foundation
 
-struct NowPlayingJson: Decodable {
-  let title: String
-  let artist: String
-  let album: String
-  let image: String
-}
-
 class NowPlayingModel: ObservableObject {
-  @Published var title = "..."
-  @Published var artist = "..."
-  @Published var album = "..."
-  @Published var image = ""
-  
-  func load() {
+  @Published var currentTrack: Track = .placeholder
+
+  @MainActor
+  func refresh() async {
+    do {
+      let newTrack = try await playingNow()
+      self.currentTrack = newTrack
+    } catch {
+      print("Could not refresh. Reason: \(error.localizedDescription)")
+    }
+  }
+
+  func playingNow() async throws -> Track {
     let url = URL(string: "https://kyan-jukebox-now-playing.deno.dev")!
-    var request = URLRequest(url: url)
-    request.httpMethod = "GET"
-    request.cachePolicy = NSURLRequest.CachePolicy.useProtocolCachePolicy
-    
-    URLSession.shared.dataTask(with: request) { data, _, error in
-      if let data = data {
-        if let decodedResponse = try? JSONDecoder().decode(NowPlayingJson.self, from: data) {
-          print("JSON fetched")
-          
-          DispatchQueue.main.async {
-            self.title = decodedResponse.title
-            self.artist = decodedResponse.artist
-            self.album = decodedResponse.album
-            self.image = decodedResponse.image
-          }
-          
-          return
-        }
-      }
-      
-      print("Fetch failed: \(error?.localizedDescription ?? "Unknown error")")
-    }.resume()
+    let (data, _) = try await URLSession.shared.data(from: url)
```

Comment:
Gah! I was so close with this when I tried. I was trying to use `URLSession.shared.dataTask(with: request)` (mainly because initially the data could only be fetched with a POST) and I just couldn't get the syntax right.

Reply:
Yeah, there is also a `URLRequest` based version (IIRC it's `data(for: URLRequest)` rather than `from:`) if we ever need non-GET requests.